test/hotspot/jtreg/compiler/c2/cr7200264/TestIntVect.java

```diff
@@ -21,622 +21,763 @@
+            System.exit(97);
+        }
+        System.out.println("PASSED");
+
+    }
+
+    int test_sum(int[] a1) {
+        int sum = 0;
+        for (int i = 0; i < a1.length; i+=1) {
+            sum += a1[i];
+        }
+        return sum;
+    }
+
+    @Test
+    @IR(counts = { IRNode.ADD_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_addc(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]+VALUE);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.ADD_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_addv(int[] a0, int[] a1, int b) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]+b);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.ADD_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_adda(int[] a0, int[] a1, int[] a2) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]+a2[i]);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.ADD_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_subc(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]-VALUE);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.SUB_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_subv(int[] a0, int[] a1, int b) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]-b);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.SUB_VI, ">= 1", },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_suba(int[] a0, int[] a1, int[] a2) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]-a2[i]);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.SUB_VI, ">= 1", IRNode.LSHIFT_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_mulc(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]*VALUE);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.SUB_VI, ">= 1", IRNode.LSHIFT_VI, ">= 1" },
+        applyIfCPUFeature = {"sse2", "true"})
+    void test_mulc_n(int[] a0, int[] a1) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]*(-VALUE));
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.MUL_VI, ">= 1" },
+        applyIfCPUFeature = {"sse4.1", "true"})
+    void test_mulv(int[] a0, int[] a1, int b) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]*b);
+        }
+    }
+
+    @Test
+    @IR(counts = { IRNode.MUL_VI, ">= 1" },
+        applyIfCPUFeature = {"sse4.1", "true"})
+    void test_mula(int[] a0, int[] a1, int[] a2) {
+        for (int i = 0; i < a0.length; i+=1) {
+            a0[i] = (int)(a1[i]*a2[i]);
+        }
+    }
+
+    void test_divc(int[] a0, int[] a1) {
```

Comment:
What about these tests without IR verification? Are they expected to be not vectorized?

Reply:
That's a good question. I based the current `@IR` verifications on the vector nodes checked in the original test, and the `test_*` functions that do not currently have `@IR` annotations are those that do not result in any of the vector operations part of the original test.

I additionally checked a few of these functions in IGV, and none of them resulted in any vector operations. I guess we can remove them altogether, unless they are actually supposed to generate vector operations.

Reply:
Thanks for checking it. Maybe @eme64 can also double check if these methods are expected to fail vectorization or if there are some missing optimization opportunities.

If the methods cannot be vectorized and we do not want to follow up on them, then I guess both is fine: leaving them in or removing them. Since it's an existing test it might be best to keep the code as it previously was.